piccolo_api/jwt_auth/middleware.py

```diff
@@ -130,6 +137,10 @@ async def __call__(self, scope, receive, send):
         """
         allow_unauthenticated = self.allow_unauthenticated
 
+        if scope["path"] in self.excluded_paths:
```

Comment:
We can enhance this quite easily to allow `*`. For example `foo/*`.

```python
for excluded_path in self.excluded_paths:
    if excluded_path.endswith('*'):
        if scope['path'].startswith(excluded_path.rstrip('*')):
            await self.asgi(scope, receive, send)
            return
    else:
        if scope['path'] == excluded_path:
            await self.asgi(scope, receive, send)
            return
```

It would be a really useful feature.
        
```

Reply:
Great idea. With this we can exclude the entire paths of the app, for example for `private_app` we can exclude all paths like `private/*`. I only need to change this line 
```python
if scope['path'].startswith(excluded_path.rstrip('*')):
``` 
to 
```python
if excluded_path.startswith(excluded_path.rstrip("*")):
``` 
to have an effect.